network/gossip/libp2p/network.go

```diff
@@ -66,9 +74,15 @@ func NewNetwork(
 		top:             top,
 		metrics:         metrics,
 		subscriptionMgr: newSubscriptionManager(mw),
+		state:           state,
 	}
 
-	o.SetIDs(ids)
+	// seed the network with the appropriate ids as per the epoch state
+	ids, err := idsFromState(state)
```

Comment:
@jordanschalm - now have this function `idsFromState` which gets the ids based on the epoch phase